src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Models/Provider/WhenAddingDraftApprenticeship.cs

```diff
@@ -0,0 +1,141 @@
+using System;
+using System.Linq;
+using AutoFixture;
+using FluentAssertions;
+using NUnit.Framework;
+using SFA.DAS.CommitmentsV2.Domain.Entities;
+using SFA.DAS.CommitmentsV2.Messages.Events;
+using SFA.DAS.CommitmentsV2.Models;
+using SFA.DAS.CommitmentsV2.Types;
+using SFA.DAS.Testing.Builders;
+using SFA.DAS.UnitOfWork;
+
+namespace SFA.DAS.CommitmentsV2.UnitTests.Models.Provider
+{
+    [TestFixture]
+    [Parallelizable]
+    public class WhenAddingDraftApprenticeship
+    {
+        private AddDraftApprenticeshipTestFixture _fixture;
+        
+        [SetUp]
+        public void Arrange()
+        {
+            _fixture = new AddDraftApprenticeshipTestFixture();
+        }
+
+        [Test]
+        public void ThenShouldAddDraftApprenticeshipToCohort()
+        {
+            var draftApprenticeship = _fixture.AddDraftApprenticeship();
+
+            draftApprenticeship.Should().BeSameAs(_fixture.Cohort.Apprenticeship.SingleOrDefault());
+        }
+
+        [Test]
+        public void ThenShouldPublishDraftApprenticeshipCreatedEvent()
+        {
+            var draftApprenticeship = _fixture.AddDraftApprenticeship();
+
+            _fixture.UnitOfWorkContext.GetEvents().Should().HaveCount(1)
+                .And.Subject.Cast<DraftApprenticeshipCreatedEvent>().Single().Should().BeEquivalentTo(new DraftApprenticeshipCreatedEvent(
+                    cohortId: _fixture.Cohort.Id,
+                    draftApprenticeshipId: draftApprenticeship.Id,
+                    uln: _fixture.DraftApprenticeshipDetails.Uln,
+                    reservationId: draftApprenticeship.ReservationId.Value,
+                    createdOn: draftApprenticeship.CreatedOn.Value));
+        }
+
+        [Test]
+        public void ThenShouldReturnDraftApprenticeship()
+        {
+            var draftApprenticeship = _fixture.AddDraftApprenticeship();
+
+            draftApprenticeship.Should().NotBeNull().And.Match<DraftApprenticeship>(d =>
+                d.FirstName == _fixture.DraftApprenticeshipDetails.FirstName &&
+                d.LastName == _fixture.DraftApprenticeshipDetails.LastName &&
+                d.Uln == _fixture.DraftApprenticeshipDetails.Uln &&
+                d.ProgrammeType == _fixture.DraftApprenticeshipDetails.TrainingProgramme.ProgrammeType &&
+                d.CourseCode == _fixture.DraftApprenticeshipDetails.TrainingProgramme.CourseCode &&
+                d.Cost == _fixture.DraftApprenticeshipDetails.Cost &&
+                d.StartDate == _fixture.DraftApprenticeshipDetails.StartDate &&
+                d.EndDate == _fixture.DraftApprenticeshipDetails.EndDate &&
+                d.DateOfBirth == _fixture.DraftApprenticeshipDetails.DateOfBirth &&
+                d.ReservationId == _fixture.DraftApprenticeshipDetails.ReservationId);
+        }
+
+        [Test]
+        public void AndEditStatusIsProviderOnlyAndPartyIsProviderThenShouldReturnDraftApprenticeship()
```

Comment:
Do we need a test to cover the case when the EditStatus is not with the Provider? And look for the domain exception. (Ignore you do this in the Validation Tests section